### ISSUE TITLE
Mongo version never collected

### ISSUE THREAD

**Issue by gsanchezgavier** [open-telemetry/opentelemetry-collector-contrib#27441]
### Component(s)

receiver/mongodb

## Description
After this bugfix #23860 version is never collected.
## Steps to Reproduce
Configure the receiver to scrape a v4.4 mongo instance.
## Expected Result
This metric [not being collected](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.86.0/receiver/mongodbreceiver/metrics.go#L129C2-L129C2) because of the mongo version
## Actual Result
Error logged 
```
 Error scraping metrics    {"kind": "receiver", "name": "mongodb", "data_type": "metrics", "error": "failed to collect metric mongodb.extent.count with attribute(s) admin: could not find key for metric; failed to fetch index stats metrics: (Unauthorized) not authorized on admin to execute command... 
```

### Collector version

v0.86.0

## Environment
mongodb v4.4

### OpenTelemetry Collector configuration

```yaml
mongodb:
    hosts:
      - endpoint: localhost:27017
    username: root
    password: fake
    tls:
      insecure: true
      insecure_skip_verify: true
```

### Additional context

I'm submitting the fix PR

**Comment by crobert-1** [comment]
I checked the PR, I agree this is a bug. A code owner has also commented on the PR, so I'm removing `needs triage`.

**Comment by crobert-1** [comment]
Fixed by #27442
